nova_vm/src/engine/bytecode/bytecode_compiler.rs

```diff
@@ -48,7 +53,26 @@ pub(crate) enum NamedEvaluationParameter {
     ReferenceStack,
 }
 
-pub(crate) struct CompileContext<'agent, 'gc, 'scope> {
+pub(crate) struct JumpTarget {
+    /// Depth of the lexical of the jump target.
+    depth: u32,
+    /// `continue;` statements that target this jump target.
+    pub(crate) continues: Vec<JumpIndex>,
+    /// `break;` statements that target this jump target.
+    pub(crate) breaks: Vec<JumpIndex>,
+}
+
+impl JumpTarget {
+    pub(super) fn new(depth: u32) -> Self {
+        Self {
+            depth,
+            continues: vec![],
+            breaks: vec![],
+        }
+    }
+}
+
+pub(crate) struct CompileContext<'agent, 'script, 'gc, 'scope> {
```

Comment:
There's getting to be a lot of lifetimes here, some docs could be helpful

Reply:
Good point; I'll add some.